hera_sim/visibilities/vis_cpu.py

```diff
@@ -315,38 +227,48 @@ def _base_simulate(self, crd_eq, I_sky):
             myid = self.mpi_comm.Get_rank()
             nproc = self.mpi_comm.Get_size()
 
+        if self.correct_source_positions:
+            # TODO: check if this is the right time to be using...
+            ra, dec = self.correct_point_source_pos(
+                data_model, obstime=data_model.uvdata.time_array[0]
+            )
+        else:
+            ra, dec = data_model.sky_model.ra, data_model.sky_model.dec
+
+        crd_eq = convs.point_source_crd_eq(ra, dec)
+
         # Convert equatorial to topocentric coords
-        eq2tops = self.get_eq2tops()
+        eq2tops = self.get_eq2tops(data_model.uvdata, data_model.lsts)
 
         # Get pixelized beams if required
         if self.use_pixel_beams:
-            beam_lm = self.get_beam_lm()
+            beam_lm = self.get_beam_lm(data_model)
         else:
             beam_list = [
-                self.beams[np.where(self.beam_ids == ant)[0][0]]
-                for ant in self.ant_list
+                data_model.beams[np.where(data_model.beam_ids == ant)[0][0]]
+                for ant in data_model.ant_list
```

Comment:
This isn't going to work with the way the `beam_ids` array is constructed, since `beam_ids` won't necessarily have a unique entry for each antenna.

I think I'm a little too tired (and my head too stuck in the mist of edge cases) to figure out the right way to do this right now, but the thing I'm worrying about is the case where the antenna numbers don't monotonically increase in the `UVData` object. I'm mainly concerned about ensuring that we get the ordering right on all the moving parts here, so that the beams get matched with the right antennas and the visibilities returned by the simulator match the blt-ordering imposed on the data array.

Reply:
Yup. I'm with you. I'm wary of all of this, because I know there's a few moving parts. Perhaps we should chat live about it, and maybe rope in @mkolopanis  / @bhazelton .

Reply:
I think a live chat would be good. Maybe after next week's Validation meeting?

Reply:
Sounds good.